packages/esm-service-queues-app/src/add-provider-queue-room/add-provider-queue-room.component.tsx

```diff
@@ -33,110 +32,125 @@ import {
-    localStorage.setItem('queueLocationName', selectedItem.name);
-    updateSelectedQueueLocationName(selectedItem.name);
-    updateSelectedQueueLocationUuid(selectedItem.id);
-  };
-
-  const handleRetainLocation = (e) => {
-    localStorage.setItem('isPermanentProviderQueueRoom', JSON.parse(e.target.checked));
-    updateIsPermanentProviderQueueRoom(JSON.parse(e.target.checked));
-  };
-
-  const onSubmit = useCallback(() => {
-    if (!queueRoomUuid) {
-      setIsMissingQueueRoom(true);
-      return;
-    }
-    setIsMissingQueueRoom(false);
-
-    if (providerRoom?.length > 0) {
-      updateProviderToQueueRoom(queueProviderMapUuid, queueRoomUuid, providerUuid).then(
-        () => {
-          showSnackbar({
-            isLowContrast: true,
-            title: t('updateRoom', 'Update room'),
-            kind: 'success',
-            subtitle: t('queueRoomUpdatedSuccessfully', 'Queue room updated successfully'),
-          });
-          closeModal();
-          localStorage.setItem('lastUpdatedQueueRoomTimestamp', new Date().toString());
-          updatedSelectedQueueRoomTimestamp(new Date());
-          mutate();
-        },
-        (error) => {
-          showSnackbar({
-            title: t('queueRoomAddFailed', 'Error adding queue room'),
-            kind: 'error',
-            isLowContrast: false,
-            subtitle: error?.message,
-          });
-          closeModal();
-        },
-      );
-    } else {
-      addProviderToQueueRoom(queueRoomUuid, providerUuid).then(
-        () => {
-          showSnackbar({
-            isLowContrast: true,
-            title: t('addRoom', 'Add room'),
-            kind: 'success',
-            subtitle: t('queueRoomAddedSuccessfully', 'Queue room added successfully'),
-          });
-          closeModal();
-          localStorage.setItem('lastUpdatedQueueRoomTimestamp', new Date().toString());
-          updatedSelectedQueueRoomTimestamp(new Date());
-          mutate();
-        },
-        (error) => {
-          showSnackbar({
-            title: t('queueRoomAddFailed', 'Error adding queue room'),
-            kind: 'error',
-            isLowContrast: false,
-            subtitle: error?.message,
-          });
-        },
-      );
+  }, []);
+
+  const handleQueueLocationChange = useCallback(
+    ({ selectedItem }) => {
+      if (!selectedItem) return;
+
+      localStorage.setItem('queueLocationUuid', selectedItem.id);
+      localStorage.setItem('queueLocationName', selectedItem.name);
+      updateSelectedQueueLocationName(selectedItem.name);
+      updateSelectedQueueLocationUuid(selectedItem.id);
+      setValue('queueLocationUuid', selectedItem.id);
+    },
+    [setValue],
+  );
+
+  const handleRetainLocation = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const isChecked = event.target.checked;
+      localStorage.setItem('isPermanentProviderQueueRoom', String(isChecked));
+      updateIsPermanentProviderQueueRoom(isChecked);
```

Comment:
```suggestion 
      const isChecked = event.target.checked;
      const isPermanent = String(isChecked) === 'true';
      localStorage.setItem('isPermanentProviderQueueRoom', String(isChecked));
      updateIsPermanentProviderQueueRoom(isPermanent);
```

Let's standardize the type handling for localStorage